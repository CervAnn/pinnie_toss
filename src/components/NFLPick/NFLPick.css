.NFLPick {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}

#nfl-team-badge {
  height: 200px;
  width: 200px;
  object-fit: cover;
}

#team-name-badge {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

#nfl-team-name {
  color: white;
  font-size: 35px;
  text-shadow: -1px -1px 0 #193d7b, 1px -1px 0 #193d7b, -1px 1px 0 #193d7b,
    1px 1px 0 #193d7b;
}
