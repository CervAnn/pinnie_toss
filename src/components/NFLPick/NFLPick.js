import React from 'react'
import { connect } from 'react-redux'
import '../NFLPick/NFLPick.css'


export const NFLPick = ({randomTeam}) => {
  return (
    <section className="NFLPick">
      <article id="team-name-badge">
        <img id="nfl-team-badge" src={randomTeam.strTeamBadge} alt=""/>
        <h1>{randomTeam.strTeam}</h1>
      </article>
    </section>
  )
}

export const mapStateToProps = store => ({
  randomTeam: store.randomTeam
})

export default connect(mapStateToProps)(NFLPick);