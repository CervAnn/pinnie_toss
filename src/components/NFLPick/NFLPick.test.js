import React from "react";
import { shallow } from "enzyme";
import { NFLPick, mapStateToProps } from "./NFLPick";

describe("NFLPick", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <NFLPick
        teams={[]}
        randomTeam={{
          idTeam: "134920",
          strLeague: "NFL",
          strTeam: "New England Patriots",
          strTeamBadge:
            "https://www.thesportsdb.com/images/media/team/badge/xtwxyt1421431860.png"
        }}
      />
    );
  });

  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("mapStateToProps", () => {
    it("should return an object with the contents of the store", () => {
      let mockStore = {
        teams: [
          {
            idTeam: "134920",
            strLeague: "NFL",
            strTeam: "New England Patriots",
            strTeamBadge:
              "https://www.thesportsdb.com/images/media/team/badge/xtwxyt1421431860.png"
          },
          {
            idTeam: "134922",
            strLeague: "NFL",
            strTeam: "Baltimore Ravens",
            strTeamBadge:
              "https://www.thesportsdb.com/images/media/team/badge/einz3p1546172463.png"
          }
        ],
        randomTeam: {
          idTeam: "134920",
          strLeague: "NFL",
          strTeam: "New England Patriots",
          strTeamBadge:
            "https://www.thesportsdb.com/images/media/team/badge/xtwxyt1421431860.png"
        }
      };

      let expected = {
        randomTeam: {
          idTeam: "134920",
          strLeague: "NFL",
          strTeam: "New England Patriots",
          strTeamBadge:
            "https://www.thesportsdb.com/images/media/team/badge/xtwxyt1421431860.png"
        }
      };

      expect(mapStateToProps(mockStore)).toEqual(expected);
    });
  });
});
