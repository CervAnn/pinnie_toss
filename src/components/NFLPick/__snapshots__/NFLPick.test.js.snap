// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NFLPick should render without crashing 1`] = `
<section
  className="NFLPick"
>
  <article
    id="team-name-badge"
  >
    <img
      alt=""
      id="nfl-team-badge"
      src="https://www.thesportsdb.com/images/media/team/badge/xtwxyt1421431860.png"
    />
    <h1>
      New England Patriots
    </h1>
  </article>
</section>
`;
